### DISCUSSION TITLE
System Requirement for High Load Environment

### DISCUSSION THREAD

**Discussion by bytefact** [chatwoot/chatwoot#2593] [Self hosted] [answered]
Hi guys, 

Can you please suggest system requirements and necessary system configuration for a high load environment? 

After deployment, we are going to receive ~130 chat request per minute. Below is the chat request distribution -
1. Website 1:  15 request per min.
2. Website 2:  15 request per min.
3. Facebook Page 1: 30 request per min.
4. Facebook Page 2: 70 Request per min.

Thanks.

**Comment by sojan-official** [comment, marked as answer]
Apart from the RPS, the performance of chatwoot installation could also depend on the number of online users on your website who stays connected to chatwoot via WebSockets. 

So I would recommend a deployment set-up where you could scale up the number of web and worker servers individually. 
example: Heroku, Kubernetes, VMs with a load balancer etc
ref: https://www.chatwoot.com/docs/self-hosted/deployment/architecture 

Start with a couple of servers for web and workers and fine-tune the set-up based on how the system behaves.
